setup.py

```diff
@@ -1,28 +1,72 @@
 #!/usr/bin/env python3
 
+import os
 import sys
 import numpy as np
 
-from setuptools import setup
-from distutils.extension import Extension
-from Cython.Build import cythonize
+from setuptools.extension import Extension
+from Cython.Build import build_ext
+
+dirname = os.path.dirname(os.path.abspath(__file__))
+
+if "--no-build-core" in sys.argv:
+    build_core = False
+    sys.argv.remove("--no-build-core")
+else:
+    build_core = os.path.exists(os.path.join(dirname, "primitiv-core/CMakeLists.txt"))
+
+if build_core:
+    from skbuild import setup
+else:
+    from setuptools import setup
+
+bundle_core_library = False
+if "--bundle-core-library" in sys.argv:
+    if not build_core:
+        print("primitiv-core/CMakeLists.txt is not found", file=sys.stderr)
```

Comment:
"primitiv-core/CMakeLists.txt" exists twice. Better to use some variables.

Reply:
Fixed